src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,225 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(final Car car) {
+        String query = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query,
+                         Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create car: " + car + ".", e);
+        }
+        if (car.getId() != null) {
+            addDrivers(car.getId(), car.getDrivers());
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(final Long carId) {
+        Optional<Car> carResult = Optional.empty();
```

Comment:
```suggestion
        Сar carResult = null;
```
Why do we need Optional at this stage? We only need it as a return type. Better just use Optional.ofNullable() at the end

Reply:
Done